requests/adapters.py

```diff
@@ -464,7 +466,9 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                         pool=conn,
                         connection=low_conn,
                         preload_content=False,
-                        decode_content=False
+                        decode_content=False,
+                        enforce_content_length=True,
+                        request_method=request.method
```

Comment:
Alright, the method is available inside of `from_httplib` but only as a private variable `_method` on the `httplib` object. I don't know if we want to use that inside of urllib3, so this would be the alternative option. `from_httplib` is a lower level method than `urlopen`, so we should probably be providing the method like this in the `response_kw` list, rather than trying to reimplement functionality that's already happening at a higher level.